portal-impl/src/com/liferay/portal/util/PortalInstances.java

```diff
@@ -475,7 +473,8 @@ public static void removeCompany(long companyId) {
 		}
 
 		_companyIds.remove(companyId);
-		_webIds = null;
+
+		_webIds.clear();
```

Comment:
@yuhai You should not do a clear here, you need to do same thing as _companyIds.
The old logic is _webIds is populated lazily, with your change, _webIds is populated eagerly.